test/v5/kyberNetwork.js

```diff
@@ -0,0 +1,1217 @@
+
+    Helper.assertEqual(indices.length, sells.length, "bad sells array size");
+    Helper.assertEqual(indices.length, buys.length, "bad buys array size");
+
+    await pricing.setCompactData(buys, sells, block, indices, {from: operator});
+    return pricing;
+}
+
+async function addReservesToNetwork(networkInstance, reserveInstances) {
+    for (const [key, value] of Object.entries(reserveInstances)) {
+        reserve = value;
+        console.log("adding reserve type: " + reserve.type + " address: " + reserve.address + " pricing: " + reserve.pricing);
+        networkInstance.addReserve(reserve.address, reserve.reserveId, reserve.isFeePaying, reserve.address, {from: operator});
+        for (let j = 0; j < numTokens; j++) {
+            networkInstance.listPairForReserve(reserve.address, tokens[j].address, true, true, true, {from: operator});
+        }
+    }
+}
+
+async function removeReservesFromNetwork(networkInstance, reserveInstances) {
+    for (const [key, value] of Object.entries(reserveInstances)) {
+        reserve = value;
+        console.log("removing reserve type: " + reserve.type + " address: " + reserve.address + " pricing: " + reserve.pricing);
+        for (let j = 0; j < numTokens; j++) {
+            networkInstance.listPairForReserve(reserve.address, tokens[j].address, true, true, false, {from: operator});
+        }
+        networkInstance.rmReserve(reserve.address, {from: operator});
+    }
+}
+
+function genReserveID(reserveID, reserveAddress) {
+    return reserveID + reserveAddress.substring(2,10);
+}
+
+async function fetchReservesRatesFromNetwork(networkInstance, tokenAddress, qty, isTokenToEth) {
+    reservesArray = [];
+    result = await networkInstance.getPricesForToken(tokenAddress, qty);
+    if (isTokenToEth) {
+        reserves = result.sellReserves;
+        rates = result.sellRates;
+    } else {
+        reserves = result.buyReserves;
+        rates = result.buyRates;
+    }
+    for (i=0; i<reserves.length; i++) {
+        reserveAddress = reserves[i];
+        reserve = reserveInstances[reserveAddress];
+        reserve.rate = rates[i];
+        reservesArray.push(reserve);
+    }
+    return reservesArray;
+}
+
+async function assertSameEtherBalance(accountAddress, expectedBalance) {
+    let balance = await Helper.getBalancePromise(accountAddress);
+    Helper.assertEqual(balance, expectedBalance, "wrong ether balance");
+}
+
+async function assertSameTokenBalance(accountAddress, token, expectedBalance) {
+    let balance = await token.balanceOf(accountAddress);
+    Helper.assertEqual(balance, expectedBalance, "wrong token balance");
+}
+
+//returns random integer between min (inclusive) and max (inclusive)
+function getRandomInt(min, max) {
+    min = Math.ceil(min);
+    max = Math.floor(max);
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+async function getBestReserveAndRate(reserves, src, dest, srcAmount, takerFeeBps) {
+    bestReserveData = {
+        address
```

Comment:
better have inputs here.
like for split can optionally give split values and num reserves.
for mask out mask in give percentage of reserves.
we might want to test mask out with all reserves
or mask in with no reserve